### PR TITLE
Add tests to increase coverage in the qi module

### LINKED ISSUE #299: The tools in qiskit/tools/qi/ need more testing
The files under `qiskit/tools/qi/` are partially covered by the current testsuite. (see #221)

file | statements | missing | coverage
-- | -- | -- | --
qiskit/tools/qi/qi.py | 207 | 99 | 52%
qiskit/tools/qi/pauli.py | 155 | 38 | 75%

It would be great to have more tests to cover the methods and functions in these files. Specially `qiskit/tools/qi/qi.py`.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit increases the coverage for the qiskit.tools.qi.qi module. It
adds tests for the functions missing them.

### Details and comments

Fixes #299

### CODE REVIEW

File: test/python/test_qi.py [hunk 1]
```diff
@@ -134,6 +146,117 @@ def test_concurrence(self):
                      concurrence(rho4) == 0.5)
         self.assertTrue(test_pass)
 
+    def test_concurrence_not_two_qubits(self):
```
Comment (ajavadia) [inline test/python/test_qi.py]: This actually reveals that maybe the `concurrence()` function should be updated. 

First, it should only be defined for 2 qubits. That is, the Exception message should be updated to reflect this: ("Concurence is only defined for more than two qubits").

Second, we should update the docstring to say the argument can both be a pure state and a density matrix.
```
    Args:
        state (np.array): a quantum state (1x4 array) or a density matrix (4x4) array
```

Comment (mtreinish) [reply test/python/test_qi.py]: Done

**Comment by delapuente** [comment]
@ajavadia, we have this and another PR (#779) related to testing coverage. I'm assigning them to you but feel free to unassing or attend when you need.

**Comment by jaygambetta** [comment]
Can we approve this its a simple extensions.

**Pull request merged**